### ISSUE TITLE
Asking for user confirmation fails when using SSH key

### ISSUE THREAD

**Issue by pentamassiv** [keepassxreboot/keepassxc#8471]
## Overview
I want to store my ED25519 SSH keys in KeepassXC as an attachment and be asked for confirmation before they are used, but that option breaks it.

## Steps to Reproduce
1. Set up [SSH with Github](https://docs.github.com/en/authentication/connecting-to-github-with-ssh)
2. Confirm the identity is not present by running `ssh-add -l`
3. Add ED25519 SSH keys and passphrase to KeepassXC, enable everything for the SSH agent
4. Select the option "Require user confirmation when this key is used" for the key
5. Confirm the identity was added by running `ssh-add -l`
6. Try testing it with Github:
```
[user@host ~]$ ssh -T git@github.com
sign_and_send_pubkey: signing failed for ED25519 "user@domain.com" from agent: agent refused operation
git@github.com: Permission denied (publickey).
```
7. Click "Remove from agent"
8. Confirm the identity is not present by running `ssh-add -l`
9. Deselect the option "Require user confirmation when this key is used"
10. Click "Add to agent"
11. Confirm the identity was added by running `ssh-add -l`
12. Test it with Github again
```
ssh -T git@github.com
Hi user! You've successfully authenticated, but GitHub does not provide shell access.
```
## Expected Behavior
Step 6 should have resulted in the same outcome as step 12 after I confirmed the usage of the key

## Actual Behavior
Step silently fails and I get this error from which it is hard to guess what is wrong

## Context
KeePassXC - Version 2.7.1
Revision: 5916a8f

Qt 5.15.6
Debugging mode is disabled.

Operating system: Arch Linux
CPU architecture: x86_64
Kernel: linux 5.19.7-arch1-1

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- Botan 2.19.2

Desktop Env: Gnome
Windowing System: Wayland

Seahorse is also installed and enabled.

**Comment by droidmonkey** [comment]
At step 5 KeePassXC is done. Since the key was successfully added we no longer have a role in the _use of_ that key. You need to file a bug with the ssh-add program, not us.

**Comment by dryya** [comment]
For the record, I actually don't think there's any bug in openssh/ssh-agent: I ran into this issue and fixed by enabling a custom ssh_askpass which works on wayland. It's laid out very well [in this blog post](https://spencerkrum.com/posts/ssh-askpass-wayland/); big thanks to the author, Spencer Krum. 

In summary: set SSH_ASKPASS=~/.local/bin/ssh-askpass (or wherever you'd like to place the script) on your ssh-agent process and create that script with your favorite prompter. I used the author's recommendation as follows:

`/usr/bin/zenity --title ssh-askpass --question --text "A PROGRAM REQUIRES A KEY FROM THE SSH AGENT. \n Approve?"`
